proposals/buildfile.md

```diff
@@ -0,0 +1,118 @@
+# Jib CLI Buildfile Specification
+
+Specification for a buildfile describing building a container image. This buildfile can be
+used by the jib-cli to generate a container. It is translated directly into a buildplan and
+passed to the builder.
+
+```
+# "FROM" with option to use os/architecture for manifest lists
+baseImage: "gcr.io/distroless/java:8"
+osHint: linux
+architectureHint: amd64
+
+creationTime: 0 # millis since epoch or iso8601 creation time
+format: Docker # Docker or OCI
+
+environment:
+  "MY_KEY1": "value"
+  "MY_KEY2": "value"
+labels:
+  "com.example.owner": "person"
+  "com.example.mode": "dev"
+volumes:
+  - "/myvolume"
+  - "/youvolume"
+
+exposedPorts:
+  - "123/udp"
+  - "456"
+  - "789/tcp"
+
+user: root
+workingDirectory: "/somewhere"
+entrypoint:
+  - "java"
+  - "-jar"
+cmd:
+  - "myjar.jar"
+
+# global file properties applied to all layers
+fileProperties:
+  filePermissions: "644"
+  directoryPermissions: "755"
+  user: "0"
+  group: "0"
+  timestamp: "0"
+layers:
```

Comment:
The build plan allows adding a layer archive tarball. Perhaps Jib CLI can support that too?

Reply:
yeah i'm looking into how to support that parsing with the yaml parser I'm using